src/spreadsheets.js

```diff
@@ -0,0 +1,286 @@
+        // this may be significantly larger than the used cells without incurring any computational cost
+    },
+    components: {
+        apiClient: { // the authorized API client
+            type: 'spreadsheets.sheetsAPIClient'
+        }
+    },
+    // TODO: ask A about old-style event passing versus current setup, 
+    // which is an onCreate listener triggering an invoker 
+    // '{that}.apiClient.events.onResolve': 'spreadsheets.cell.updateValue',
+    // '{that}.apiClient.events.onError': 'spreadsheets.cell.displayError'
+    listeners: { // read the remote spreadsheet on component creation
+        'onCreate.getRemoteContent': 'spreadsheets.spreadsheet.getRemoteContent({that}, {apiClient})'
+    },
+    // When the model is written to, update the remote sheet
+    // TODO: where is the change source stored?
+    modelListeners: {
+        '*': {
+            func: 'spreadsheets.spreadsheet.setRemoteContent',
+            args: ['{that}', '{apiClient}', '{change}.path.0', '{change}.value'],
+            // FIXME: currently, this model relay doesn't work as intended. 
+            // It should systematically exclude mutations coming from reads. 
+            // Can those be associated with a particular invoker on the apiClient sub-component or locally? 
+            // The kludge I can make right now is to check the written value against the last-read value.
+            // excludeSource: 'init' does not seem to be what I want, because multiple reads could happen over the component life cycle.
+        }
+    }
+});
+
+// TODO: refactor this messy function
+// TODO: I wonder if I can more declaratively specify
+//   which api call with which body
+//   how to decode and extract data
+//   how to fill the model
+// and whether that has value for the design
+spreadsheets.spreadsheet.getRemoteContent = function(that, apiClient) {
+    var metadataPromise = apiClient.getSpreadsheet(apiClient, that.options.spreadsheetId);
+    var requests = [];
+    metadataPromise.then(function(res) {
+        fluid.each(res.data.sheets, function(sheetObject) {
+            var {title} = sheetObject.properties;
+            var {columnCount, rowCount} = sheetObject.properties.gridProperties;
+            requests.push({title, columnCount, rowCount});
+        });
+        var ranges = [];
+        fluid.each(requests, function({title, columnCount, rowCount}) {
+            var boundingRange = title + '!' + 'A1:' + spreadsheets.coordToA1(columnCount, rowCount);
+            that.sheetBounds[title] = boundingRange;
+            ranges.push(boundingRange);
+        });
+        var valuePromise = apiClient.getRanges(
+            apiClient,
+            that.options.spreadsheetId,
+            ranges);
+        valuePromise.then(function(res) {
+            fluid.each(res.data.valueRanges, function({range, values}) {
+                var sheetName = range.split('!')[0];
+                that.applier.change(sheetName, values);
```

Comment:
May as well apply these in a transaction, even though it's perhaps unlikely that someone might want to depend on atomicity of updates who is listening to stuff in multiple sheets. https://github.com/amb26/infusion/blob/FLUID-6145/tests/framework-tests/core/js/DataBindingTests.js#L410